firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -198,11 +200,12 @@ class TabManagerImplementation: NSObject, TabManager, FeatureFlaggable {
         if #available(iOS 15.4, *) {
             configuration.preferences.isElementFullscreenEnabled = true
         }
-        if isPrivate {
-            configuration.websiteDataStore = WKWebsiteDataStore.nonPersistent()
-        } else {
-            configuration.websiteDataStore = WKWebsiteDataStore.default()
-        }
+
+        // Use shared pool for regular mode, isolated pool for private mode
+        configuration.processPool = isPrivate ? WKProcessPool() : defaultProcessPool
+        configuration.websiteDataStore = isPrivate
+                ? WKWebsiteDataStore.nonPersistent()
+                : WKWebsiteDataStore.default()
```

Comment:
Nit: I think using an if instead of 2 ternary operators would read better since they are both checking the same thing.